storage/src/mmr/verification.rs

```diff
@@ -39,6 +39,37 @@ impl<H: CHasher> PartialEq for Proof<H> {
     }
 }
 
+impl<H: CHasher> EncodeSize for Proof<H> {
```

Comment:
Not really feedback on this specific PR, but shouldn't we be able to write a macro that implements these "default" types of code serialize/deserialze implementations?  This all seems very boilerplate.

Reply:
I've advocated for manually encoding to avoid silent changes from struct modification fwiw